cirq/linalg/decompositions.py

```diff
@@ -512,12 +513,15 @@ def interaction_matrix(m: np.ndarray, c: float) -> np.ndarray:
         y_mat = np.array([[0, -1j], [1j, 0]])
         z_mat = np.array([[1, 0], [0, -1]])
 
-        return self.global_phase * combinators.dot(
-            after,
-            interaction_matrix(z_mat, z),
-            interaction_matrix(y_mat, y),
-            interaction_matrix(x_mat, x),
-            before,
+        return self.global_phase * cast(
```

Comment:
ditto

(Possibly could apply to other use of `cast` below, too.)

Reply:
changed this per above. what other use of cast?

Reply:
Well, there are six more left at this point. It'd be nice to get rid of as many as you can. They may hide bugs and make code harder to read.